examples/armbench-segmentation/tensorleap.py

```diff
@@ -0,0 +1,400 @@
+import os
+from typing import Union, List
+
+import numpy as np
+import tensorflow as tf
+from PIL import Image
+
+from code_loader import leap_binder
+from code_loader.contract.enums import (
+    DatasetMetadataType,
+    LeapDataType
+)
+
+from code_loader.contract.datasetclasses import PreprocessResponse
+from pycocotools.coco import COCO
+
+from armbench_segmentation.gcs_utils import _download
+from armbench_segmentation.preprocessing import BACKGROUND_LABEL, CATEGORIES, IMAGE_SIZE, SMALL_BBS_TH, DIR, IMG_FOLDER, \
+    LOAD_UNION_CATEGORIES_IMAGES, load_set, TRAIN_SIZE, VAL_SIZE, UL_SIZE, CLASSES, MAX_BB_PER_IMAGE
+from armbench_segmentation.utils.general_utils import count_obj_masks_occlusions, \
+    count_obj_bbox_occlusions, extract_and_cache_bboxes
+from armbench_segmentation.metrics import regression_metric, classification_metric, object_metric, \
+    mask_metric, over_segmented, under_segmented, metric_small_bb_in_under_segment, non_binary_over_segmented, \
+    non_binary_under_segmented, average_segments_num_over_segment, average_segments_num_under_segmented, \
+    over_segment_avg_confidence
+from armbench_segmentation.visualizers.visualizers import gt_bb_decoder, bb_decoder, \
+    under_segmented_bb_visualizer, over_segmented_bb_visualizer
+from armbench_segmentation.visualizers.visualizers_getters import mask_visualizer_gt, mask_visualizer_prediction
+
+
+# ----------------------------------------------------data processing--------------------------------------------------
+def subset_images():
+    ann_file = os.path.join(DIR, IMG_FOLDER, "train.json")
+    fpath = _download(ann_file)
+    # initialize COCO api for instance annotations
+    train = COCO(fpath)
+    x_train_raw = load_set(coco=train, load_union=LOAD_UNION_CATEGORIES_IMAGES)
+
+    ann_file = os.path.join(DIR, IMG_FOLDER, "test.json")
+    fpath = _download(ann_file)
+    val = COCO(fpath)
+    x_val_raw = load_set(coco=val, load_union=LOAD_UNION_CATEGORIES_IMAGES)
+
+    train_size = min(len(x_train_raw), TRAIN_SIZE)
+    val_size = min(len(x_val_raw), VAL_SIZE)
+    np.random.seed(0)
+    train_idx, val_idx = np.random.choice(len(x_train_raw), train_size), np.random.choice(len(x_val_raw), val_size)
+    return [PreprocessResponse(length=train_size, data={'cocofile': train,
```

Comment:
I think it would be easier to understand like that:

`training_subset = PreprocessResponse(length=train_size, data={'cocofile': train, ...`
`validation_subset = PreprocessResponse(length=val_size, data={'cocofile': val, ...`
`return [training_subset, validation_subset]`